rest/config_test.go

```diff
@@ -440,3 +443,370 @@ func TestAutoImportEnabled(t *testing.T) {
+		configServer       = "http://127.0.0.1:4981/conf"
+		dbname             = "beer_sample"
+		defaultLogFilePath = "/var/log/sync_gateway"
+		deploymentID       = "DEPID100"
+		interfaceAddress   = "4984"
+		keypath            = "/etc/ssl/certs/key.pem"
+		log                = "Admin,Access,Auth,Bucket"
+		logFilePath        = "/var/log/sync_gateway"
+		pool               = "liverpool"
+	)
+	args := []string{
+		"sync_gateway",
+		"--adminInterface", adminInterface,
+		"--bucket", bucket,
+		"--cacertpath", cacertpath,
+		"--certpath", certpath,
+		"--configServer", configServer,
+		"--dbname", dbname,
+		"--defaultLogFilePath", defaultLogFilePath,
+		"--deploymentID", deploymentID,
+		"--interface", interfaceAddress,
+		"--keypath", keypath,
+		"--log", log,
+		"--logFilePath", logFilePath,
+		"--pool", pool,
+		"--pretty"}
+
+	err := ParseCommandLine(args, flag.ContinueOnError)
+	require.NoError(t, err, "Error reading config file")
+	conf := GetConfig()
+	assert.Equal(t, adminInterface, *conf.AdminInterface)
+	databases := conf.Databases
+	assert.Len(t, databases, 1)
+	assert.Equal(t, dbname, databases[dbname].Name)
+	assert.Equal(t, cacertpath, databases[dbname].CACertPath)
+	assert.Equal(t, certpath, databases[dbname].CertPath)
+	assert.Equal(t, keypath, databases[dbname].KeyPath)
+
+}
+
+func mockBucketConfig() BucketConfig {
+	bucket := "albums"
+	username := "Alice"
+	password := "QWxpY2U="
+
+	bucketConfig := BucketConfig{
+		Bucket:   &bucket,
+		Username: username,
+		Password: password}
+
+	return bucketConfig
+}
+
+func TestGetCredentialsFromDbConfig(t *testing.T) {
+	mockBucketConfig := mockBucketConfig()
+	dbConfig := &DbConfig{BucketConfig: mockBucketConfig}
+	username, password, bucket := dbConfig.GetCredentials()
+	assert.Equal(t, mockBucketConfig.Username, username)
+	assert.Equal(t, mockBucketConfig.Password, password)
+	assert.Equal(t, *mockBucketConfig.Bucket, bucket)
+}
+
+func TestGetCredentialsFromClusterConfig(t *testing.T) {
+	mockBucketConfig := mockBucketConfig()
+	heartbeatIntervalSeconds := uint16(10)
+	clusterConfig := &ClusterConfig{
+		BucketConfig:             mockBucketConfig,
+		DataDir:                  "/var/lib/sync_gateway/data",
+		HeartbeatIntervalSeconds: &heartbeatIntervalSeconds,
+	}
+	username, password, bucket := clusterConfig.GetCredentials()
+	assert.Equal(t, mockBucketConfig.Username, username)
+	assert.Equal(t, mockBucketConfig.Password, password)
+	assert.Equal(t, *mockBucketConfig.Bucket, bucket)
+}
+
+func TestSetMaxFileDescriptors(t *testing.T) {
+	var maxFDs uint64
+	err := SetMaxFileDescriptors(&maxFDs)
+	assert.NoError(t, err, "Error setting MaxFileDescriptors")
+
+	// Set MaxFileDescriptors
+	maxFDs = DefaultMaxFileDescriptors + 1
+	err = SetMaxFileDescriptors(&maxFDs)
+	assert.NoError(t, err, "Error setting MaxFileDescriptors")
+}
+
+func TestParseCommandLineWithMissingConfig(t *testing.T) {
+	func() { config = nil }()
+	log.Printf("TestParseCommandLineWithMissingConfig:%v", GetConfig())
```

Comment:
Remove dev-time logging.

Reply:
Removed dev-time logging. Changes are available in the latest revision.